app/views/govuk_publishing_components/components/_contents_list.html.erb

```diff
@@ -12,10 +12,22 @@
   link_classes << brand_helper.color_class
   link_classes << "govuk-link--no-underline" unless underline_links
 
+  ga4_tracking ||= false
+  ga4_data = nil
+  modules = "gem-track-click"
+  if ga4_tracking
+    modules << " ga4-link-tracker"
+    ga4_data = {
+      event_name: "navigation",
+      type: "content",
+      section: t("components.contents_list.contents", locale: :en) || ""
+    }.to_json
+  end
   local_assigns[:aria] ||= {}
   component_helper = GovukPublishingComponents::Presenters::ComponentWrapperHelper.new(local_assigns)
   component_helper.add_class("gem-c-contents-list #{brand_helper.brand_class}")
-  component_helper.add_data_attribute({ module: "gem-track-click" })
+  component_helper.add_data_attribute({ module: modules })
```

Comment:
I _think_ that the code could be slightly reduced as follows (but I haven't worked on the wrapper for while so obviously check it works!).

```Ruby
ga4_tracking ||= false
if ga4_tracking
    ga4_data = {
      event_name: "navigation",
      type: "content",
      section: t("components.contents_list.contents", locale: :en) || ""
    }.to_json
end

...

component_helper.add_data_attribute({ module: "gem-track-click" })
component_helper.add_data_attribute({ module: "ga4-link-tracker" }) if ga4_tracking
component_helper.add_data_attribute({ ga4_link: ga4_data, ga4_track_links_only: "" }) if ga4_tracking
```

Reply:
Thanks. I seem to need to keep the `ga4_data = nil` though, as I can't call `if ga4_tracking` within the `data` hash. It makes the template crash. Ruby doesn't render the HTML attribute if the value is `nil` so having the `ga4_data = nil` default seems to work.